src/CalcViewModel/GraphingCalculator/EquationViewModel.cpp

```diff
@@ -31,83 +32,62 @@ namespace CalculatorApp::ViewModel
     {
     }
 
-    EquationViewModel::EquationViewModel()
-        : m_LineColor{ nullptr }
-        , m_Expression{ "" }
-        , m_IsAnalysisUpdated{ false }
-        , m_Domain{ "" }
-        , m_Range{ "" }
-        , m_XIntercept{ "" }
-        , m_YIntercept{ "" }
-        , m_Parity{ -1 }
-        , m_PeriodicityDirection{ -1 }
-        , m_PeriodicityExpression{ "" }
-        , m_Minima{ ref new Vector<String ^>() }
-        , m_Maxima{ ref new Vector<String ^>() }
-        , m_InflectionPoints{ ref new Vector<String ^>() }
-        , m_Monotonicity{ ref new Map<String ^, String ^>() }
-        , m_VerticalAsymptotes{ ref new Vector<String ^>() }
-        , m_HorizontalAsymptotes{ ref new Vector<String ^>() }
-        , m_ObliqueAsymptotes{ ref new Vector<String ^>() }
-        , m_TooComplexFeatures{ -1 }
-        , m_AnalysisError{ 0 }
-        , m_AnalysisErrorVisible{ false }
+    EquationViewModel::EquationViewModel(GraphControl::Equation ^ equation)
+        : m_AnalysisErrorVisible{ false }
         , m_KeyGraphFeaturesItems{ ref new Vector<KeyGraphFeaturesItem ^>() }
         , m_resourceLoader{ Windows::ApplicationModel::Resources::ResourceLoader::GetForCurrentView() }
     {
-    }
-
-    void EquationViewModel::OnPropertyChanged(String ^ propertyName)
-    {
-        if (propertyName == L"IsAnalysisUpdated")
+        if (equation == nullptr)
         {
-            OnIsAnalysisUpdatedChanged();
+            throw ref new InvalidArgumentException(L"Equation cannot be null");
```

Comment:
Can we be sure an end user will never hit this?

Reply:
Currently, this object is only constructed in one place and we always pass an equation. If the equation becomes null for some reason then we should fix that bug instead of silently failing.

Reply:
sounds good